controllers/comments-controllers.js

```diff
@@ -1,10 +1,16 @@
+const { checkArticleExists } = require("../models/articles-models");
 const { selectCommentsByArticleId } = require("../models/comment-models");
 
 exports.getCommentsByArticleId = (req, res, next) => {
   const { article_id } = req.params;
-  selectCommentsByArticleId(article_id)
-    .then((comments) => {
+  const commentsPromises = [selectCommentsByArticleId(article_id)];
+  if (article_id) {
+    commentsPromises.push(checkArticleExists(article_id));
+  }
```

Comment:
The controller should really only be passing data any validation should really be happening in the model.